dbt/adapters/bigquery/impl.py

```diff
@@ -159,45 +169,57 @@ def query_for_existing(cls, profile, schemas, model_name=None):
         if not isinstance(schemas, (list, tuple)):
             schemas = [schemas]
 
-        conn = cls.get_connection(profile, model_name)
-        client = conn.get('handle')
+        all_relations = []
 
-        all_tables = []
         for schema in schemas:
-            dataset = cls.get_dataset(profile, schema, model_name)
-            all_tables.extend(client.list_tables(dataset))
+            all_relations.extend(cls.list_relations(
+                profile, schema, model_name))
+
+        return {relation.identifier: relation.type
+                for relation in all_relations}
+
+    @classmethod
+    def list_relations(cls, profile, schema, model_name=None):
+        connection = cls.get_connection(profile, model_name)
+        credentials = connection.get('credentials', {})
+        client = connection.get('handle')
+
+        bigquery_dataset = cls.get_dataset(profile, schema, model_name)
+        all_tables = client.list_tables(bigquery_dataset)
 
         relation_types = {
             'TABLE': 'table',
             'VIEW': 'view',
             'EXTERNAL': 'external'
         }
 
-        existing = [(table.table_id, relation_types.get(table.table_type))
-                    for table in all_tables]
-
-        return dict(existing)
-
-    @classmethod
-    def table_exists(cls, profile, schema, table, model_name=None):
-        tables = cls.query_for_existing(profile, schema, model_name)
-        exists = tables.get(table) is not None
-        return exists
+        return [cls.Relation.create(
+            project=credentials.get('project'),
+            schema=schema,
+            identifier=table.table_id,
+            type=relation_types.get(table.table_type))
+                for table in all_tables]
 
     @classmethod
-    def drop(cls, profile, schema, relation, relation_type, model_name=None):
+    def drop_relation(cls, profile, relation, model_name=None):
         conn = cls.get_connection(profile, model_name)
         client = conn.get('handle')
 
-        dataset = cls.get_dataset(profile, schema, model_name)
-        relation_object = dataset.table(relation)
+        dataset = cls.get_dataset(profile, relation._schema, model_name)
```

Comment:
is there a reason to use `_schema` and `_identifier` here instead of the property methods?